platform.md

```diff
@@ -276,10 +275,79 @@ All lifecycle phases:
 - MUST read `CNB_PLATFORM_API` from the execution environment and evaluate compatibility before attempting to parse other inputs (see [Platform API Compatibility](#platform-api-compatibility))
 - MUST give command line inputs precedence over other inputs
 
+#### `analyzer`
+The platform MUST execute `analyzer` in the **build environment**
```

Comment:
The way we define "build environment" I don't believe this is true. You could still run analyzer on the lifecycle image for example, it is not necessary that the analyze container extend from the build image.

Reply:
@ekcasey to validate mixins against all buildpack defined mixins in this phase, won't we need to be on the build image?

Reply:
I removed this and added a line to detector about validating mixins.